File: SmartInventoryFrontend/src/types/customer.ts
import { Entity } from './common'

export interface Customer extends Entity {
  company_Name: string
  customer_Code: string
  address: string
  company_Address: string
  email: string
  mobile: string
}

export interface CreateCustomerRequest {
  company_Name: string
  customer_Code: string
  address: string
  company_Address: string
  email: string
  mobile: string
}

export interface UpdateCustomerRequest {
  company_Name?: string
  customer_Code?: string
  address?: string
  company_Address?: string
  email?: string
  mobile?: string
}

export interface CustomerState {
  customers: Customer[]
  currentCustomer: Customer | null
  loading: boolean
  error: string | null
  total: number
  skip: number
  take: number
}

File: SmartInventoryAPI/Models/Entities/Customer.cs
namespace SmartInventoryAPI.Models.Entities;

public class Customer
{
    public long ID { get; set; }
    public string? Company_Name { get; set; }
    public string? Customer_Code { get; set; }
    public string? Address { get; set; }
    public bool IsDeleted { get; set; } = false;
    public DateTime Creation_Date { get; set; } = DateTime.UtcNow;
    public string? Company_Address { get; set; }
    public string? Email { get; set; }
    public string? Mobile { get; set; }
}
